tests/gradients/test_spsa_gradient.py

```diff
@@ -295,13 +307,14 @@ def test_y0_provided(self):
         tape = qml.tape.QuantumScript.from_queue(q)
         f0 = dev.execute(tape)
         n = 9
-        tapes, fn = spsa_grad(tape, strategy="forward", approx_order=1, num_directions=n, f0=f0)
+        tapes, _ = spsa_grad(tape, strategy="forward", approx_order=1, num_directions=n, f0=f0)
         # one tape per direction, the unshifted one already was evaluated above
         assert len(tapes) == n
 
     def test_independent_parameters(self):
         """Test the case where expectation values are independent of some parameters. For those
         parameters, the gradient should be evaluated to zero without executing the device."""
+        np.random.seed(42)
```

Comment:
Where is the `np.random.seed` used? We were discussing not a long time ago that it is preferred to use a numpy [Random Generator](https://numpy.org/doc/stable/reference/random/generator.html) instead, to avoid having unwanted behaviours when the user fixes the random seed.

This is however outside the scope of this PR

Reply:
SPSA is an inherently stochastic method. It uses random samples to estimate the gradient.

I think the code of SPSA should then indeed be upgraded to use a random generator?